lms/djangoapps/teams/static/teams/js/views/teams_tab.js

```diff
@@ -28,27 +57,148 @@
+                               });
+                               self.render();
+                           };
+                           teamCollection.goTo(1, {
+                               success: successCb,
+                               error: function (collection, response) {
+                                   // Topic does not exist
+                                   if (response.status === 400) {
+                                       self.topicNotFound(topicID);
+                                   }
+                                   // No teams associated with this topic
+                                   else if (response.status === 404) {
+                                       collection.totalPages = 0;
+                                       successCb(collection);
+                                   }
+                               }
+                           });
+                       }
+                       else {
+                           this.mainView = this.teamsView;
+                           this.render();
+                       }
+                    },
+
+                   /**
+                    * Set up the tabbed view and switch tabs.
+                    */
+                   goToTab: function (tab) {
+                       this.mainView = this.tabbedView;
+                       // Note that `render` should be called first so
+                       // that the tabbed view's element is set
+                       // correctly.
+                       this.render();
+                       this.tabbedView.main.setActiveTab(tab);
+                   },
+
+                   // Error handling
+
+                   routeNotFound: function (route) {
+                       this.notFoundError(
+                           interpolate(
+                               gettext('The page %(route)s could not be found.'),
+                               {route: route},
+                               true
+                           )
+                       );
+                   },
+
+                   topicNotFound: function (topicID) {
+                       this.notFoundError(
+                           interpolate(
+                               gettext('The topic %(topic)s could not be found.'),
+                               {topic: topicID},
+                               true
+                           )
+                       );
+                   },
+
+                   /**
+                    * Called when the user attempts to navigate to a
+                    * route that doesn't exist. "Redirects" back to
+                    * the main teams tab, and adds an error message.
+                    */
+                   notFoundError: function (message) {
+                       this.router.navigate('teams', {trigger: true});
+                       this.$el.prepend($('<p class="error">' + message + '</p>')); // TODO needs styling of error message
```

Comment:
No problem. Thanks for clarifying.

On 7/15/15 5:38 PM, Marco Morales wrote:
>
> In lms/djangoapps/teams/static/teams/js/views/teams_tab.js 
> <https://github.com/edx/edx-platform/pull/8859#discussion_r34732759>:
>
> > +                           interpolate(
> > +                               gettext('The topic %(topic)s could not be found.'),
> > +                               {topic: topicID},
> > +                               true
> > +                           )
> > +                       );
> > +                   },
> > +
> > +                   /**
> > +                    * Called when the user attempts to navigate to a
> > +                    * route that doesn't exist. "Redirects" back to
> > +                    * the main teams tab, and adds an error message.
> > +                    */
> > +                   notFoundError: function (message) {
> > +                       this.router.navigate('teams', {trigger: true});
> > +                       this.$el.prepend($('<p class="error">' + message + '</p>')); // TODO needs styling of error message
>
> I'm going to bring @frrrances <https://github.com/frrrances> into the 
> discussion for this error message / styling. We have an in progress UX 
> story for visual styling of these kinds of errors within the teams 
> application. If possible, it might make sense to merge this message in 
> knowing that we'll get to styling in a follow-up PR? Thanks for 
> tagging us on this!
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/edx/edx-platform/pull/8859/files#r34732759>.
>